src/utils/useFetch.ts

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable */
+import { useState, useEffect } from "react";
+import axios from "axios";
+
+export type APIResponse<T> = { statusText: string; data?: T };
+
+export interface FetchState<T> {
+  response: APIResponse<T>;
+  error?: Error;
+  isLoaded: boolean;
+}
+
+type ObjectLiteral = { [key: string]: any };
+
```

Comment:
add comments on the requirement of the function.

Reply:
@muskankhedia PTAL